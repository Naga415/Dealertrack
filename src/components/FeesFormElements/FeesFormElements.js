"use strict";
import React from "react";
import PropTypes from "prop-types";
import TextInput from "@coxautokc/fusion-ui-components/lib/TextInput";
import Checkbox from "react-bootstrap/lib/Checkbox";
import SelectInput from "@coxautokc/fusion-ui-components/lib/SelectInput";
import SelectWithGroup from "../customizedUIComponents/SelectWithGroup";
import Input from "../customizedUIComponents/Input/Input";
import IconAdd from "@coxautokc/fusion-ui-components/lib/Icons/IconAdd";
import IconDelete from "@coxautokc/fusion-ui-components/lib/Icons/IconDelete";
import Row from "@coxautokc/fusion-ui-components/lib/Row";
import Col from "@coxautokc/fusion-ui-components/lib/Col";
import "./FeesFormElements.scss";

export default function FeesFormElements(props) {
  const paidTo = [
    {
      label: null,
      value: null
    },
    {
      label: "Dealer",
      value: "Dealer"
    },
    {
      label: "State",
      value: "State"
    },
    {
      label: "Others",
      value: "Others"
    }
  ];

  const {
    grouppedFees,
    handleChangeFromForm,
    handleChangeFromInput,
    handleBlurFromInput,
    appendFee,
    deleteFee,
    quoteFees
  } = props;

  return (
    <div id="FeesFormElements">
      {quoteFees.map((qf, index) => {
        return (
          <Row key={`row-${index}`}>
            <Col md={3} sm={3}>
              <SelectWithGroup
                displayLabel={false}
                htmlId={`feesFeeType-${index}`}
                name={`feesFeeType-${index}`}
                label="Fees"
                placeholder=" "
                options={grouppedFees}
                displayPlaceholder={true}
                // eslint-disable-next-line
                onChange={e => handleChangeFromForm("feeType", e)}
                value={qf.feeId.toString()||"0"}
              />
            </Col>

            <Col md={3} sm={3}>
              <TextInput
                displayLabel={false}
                htmlId={`feesFeeName-${index}`}
                name={`feesFeeName-${index}`}
                label="Fee Name"
                placeholder=" "
                maxLength={50}
                // eslint-disable-next-line
                onChange={e => handleChangeFromForm("feeCustomName", e)}
                value={qf.feeCustomName||""}
              />
            </Col>

            <Col md={2} sm={2}>
              <Input
                displayLabel={false}
                htmlId={`feesAmount-${index}`}
                addOnClassName="fa fa-usd"
                name={`feesAmount-${index}`}
                label="Amount"
                disabled={false}
                value={qf.feeAmount}
                //eslint-disable-next-line
                onChange={handleChangeFromInput}
                onBlur={handleBlurFromInput}
              />
            </Col>

            <Col md={1} sm={1}>
              <Checkbox
                id={`feesCap-${index}`}
                name={`feesCap-${index}`}
                label="Cap"
                //eslint-disable-next-line
                onChange={e => handleChangeFromForm("capitalizedIndicator", e)}
                checked={qf.capitalizedIndicator}
              />
            </Col>

            <Col md={2} sm={2}>
              <SelectInput
                htmlId={`feesPaidTo-${index}`}
                name={`feesPaidTo-${index}`}
                label="PaidTo"
                //eslint-disable-next-line
                onChange={e => handleChangeFromForm("paidTo", e)}
                options={paidTo}
                displayLabel={false}
                displayPlaceholder={false}
                value={qf.paidTo}
              />
            </Col>

            <Col md={1} sm={1} id={`feesInteraction-${index}`}>
              {quoteFees.length !== index + 1 ? (
                <IconAdd className="deactive-add-btn" />
              ) : (
                <IconAdd className="active-add-btn" onClick={appendFee} />
              )}

              {quoteFees.length > 1 && (
                <IconDelete
                  className="active-delete-btn"
                  // eslint-disable-next-line
                  onClick={() => deleteFee(index)}
                />
              )}
            </Col>
          </Row>
        );
      })}
    </div>
  );
}

FeesFormElements.propTypes = {
  deleteFee: PropTypes.func.isRequired,
  quoteFees: PropTypes.array.isRequired,
  handleChangeFromForm: PropTypes.func.isRequired,
  handleBlurFromInput: PropTypes.func.isRequired,
  handleChangeFromInput: PropTypes.func.isRequired,
  appendFee: PropTypes.func.isRequired,
  grouppedFees: PropTypes.object.isRequired
};
