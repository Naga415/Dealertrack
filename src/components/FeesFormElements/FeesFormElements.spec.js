"use strict";
import React from "react";
import sinon from "sinon";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import FeesFormElements from "./FeesFormElements";

const handleChangeFromForm = sinon.spy();
const handleChangeFromInput = sinon.spy();
const handleBlurFromInput = sinon.spy();
const appendFee = sinon.spy();
const deleteFee = sinon.spy();

const grouppedFees = {
  "Dealer Fees": [
    "Admin",
    "CVR Fee",
    "Delivery Handling Charges",
    "Documentation Fee"
  ],
  "Government Fees": ["Additional Highway Fee", "Agent Fee", "Air Quality"],
  "Lender Fees": ["Acquisition Fee", "Convenience Fee", "Disposition Fee"]
};

const quoteFees = [
  {
    feeId: 0,
    feeCategory: "",
    feeCategoryId: 0,
    feeType: "",
    feeTypeId: 0,
    state: "",
    feeCustomName: "",
    capitalizedIndicator: [],
    feeAmount: "",
    paidTo: ""
  }
];

afterEach(() => {
  handleChangeFromForm.reset();
  handleChangeFromInput.reset();
  handleBlurFromInput.reset();
  appendFee.reset();
  deleteFee.reset();
});

describe("FeesFormElements specs", () => {
  const props = {
    grouppedFees,
    handleChangeFromForm,
    handleChangeFromInput,
    handleBlurFromInput,
    appendFee,
    deleteFee,
    quoteFees
  };

  it("renders <FeesFormElements />", () => {
    const component = renderer.create(<FeesFormElements {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('triggers appendFee function when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const addIcon = component.find('.active-add-btnsvgicon');

    expect(addIcon).toHaveLength(1);
    addIcon.simulate('click');
    expect(appendFee.calledOnce).toEqual(true);
  });

  it('triggers change event on Fee Type when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const select = component.find('#feesFeeType-0');

    select.simulate('click');
    expect(handleChangeFromForm.calledOnce).toEqual(false);
    select.simulate('change');
    expect(handleChangeFromForm.calledOnce).toEqual(true);
    expect(handleChangeFromForm.calledWith('feeType')).toEqual(true);
  });

  it('triggers change event on Fee Name when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const fusionInput = component.find('#feesFeeName-0');

    fusionInput.simulate('click');
    expect(handleChangeFromForm.calledOnce).toEqual(false);
    fusionInput.simulate('change');
    expect(handleChangeFromForm.calledOnce).toEqual(true);
    expect(handleChangeFromForm.calledWith('feeCustomName')).toEqual(true);
  });

  it('triggers change event on Fee Cap when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const checkbox = component.find('#feesCap-0');

    checkbox.simulate('click');
    expect(handleChangeFromForm.calledOnce).toEqual(false);
    checkbox.simulate('change');
    expect(handleChangeFromForm.calledOnce).toEqual(true);
    expect(handleChangeFromForm.calledWith('capitalizedIndicator')).toEqual(true);
  });

  it('triggers change event on paidTo when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const paidToSelect = component.find('#feesPaidTo-0');

    paidToSelect.simulate('click');
    expect(handleChangeFromForm.calledOnce).toEqual(false);
    paidToSelect.simulate('change');
    expect(handleChangeFromForm.calledOnce).toEqual(true);
    expect(handleChangeFromForm.calledWith('paidTo')).toEqual(true);
  });

  it('triggers change event on amount when click', () => {
    const component = mount(<FeesFormElements {...props} />);
    const amountInput = component.find('#feesAmount-0');

    amountInput.simulate('change');
    expect(handleChangeFromInput.calledOnce).toEqual(true);
    expect(handleBlurFromInput.calledOnce).toEqual(false);

    amountInput.simulate('blur');
    expect(handleChangeFromInput.calledOnce).toEqual(true);
    expect(handleBlurFromInput.calledOnce).toEqual(true);
  });
});
