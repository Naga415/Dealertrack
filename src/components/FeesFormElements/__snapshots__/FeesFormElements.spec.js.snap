// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeesFormElements specs renders <FeesFormElements /> 1`] = `
<div
  id="FeesFormElements"
>
  <div
    className="row"
    id={undefined}
  >
    <div
      className="col-md-3 col-sm-3"
      id={undefined}
    >
      <div
        className="form-group"
      >
        <select
          className="selectInput form-control"
          disabled={undefined}
          id="feesFeeType-0"
          name="feesFeeType-0"
          onBlur={[Function]}
          onChange={[Function]}
          type={undefined}
          value="0"
        >
          <option
            value=""
          >
             
          </option>
          <optgroup
            label="Dealer Fees"
          >
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
          </optgroup>
          <optgroup
            label="Government Fees"
          >
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
          </optgroup>
          <optgroup
            label="Lender Fees"
          >
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
            <option
              label={undefined}
              value={undefined}
            />
          </optgroup>
        </select>
        
      </div>
    </div>
    <div
      className="col-md-3 col-sm-3"
      id={undefined}
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="feesFeeName-0"
          maxLength={50}
          name="feesFeeName-0"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=" "
          required={undefined}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="col-md-2 col-sm-2"
      id={undefined}
    >
      <div
        className="form-group"
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-addon"
          >
            <i
              className="fa fa-usd"
            />
          </span>
          <input
            className="form-control"
            disabled={false}
            id="feesAmount-0"
            name="feesAmount-0"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="col-md-1 col-sm-1"
      id={undefined}
    >
      <div
        className="checkbox"
        style={undefined}
      >
        <label
          title=""
        >
          <input
            checked={Array []}
            disabled={false}
            id="feesCap-0"
            label="Cap"
            name="feesCap-0"
            onChange={[Function]}
            type="checkbox"
          />
        </label>
      </div>
    </div>
    <div
      className="col-md-2 col-sm-2"
      id={undefined}
    >
      <div
        className="form-group"
      >
        <select
          className="selectInput form-control"
          disabled={undefined}
          id="feesPaidTo-0"
          name="feesPaidTo-0"
          onBlur={[Function]}
          onChange={[Function]}
          type={undefined}
          value=""
        >
          <option
            value={null}
          />
          <option
            value="Dealer"
          >
            Dealer
          </option>
          <option
            value="State"
          >
            State
          </option>
          <option
            value="Others"
          >
            Others
          </option>
        </select>
        
      </div>
    </div>
    <div
      className="col-md-1 col-sm-1"
      id="feesInteraction-0"
    >
      <svg
        className="active-add-btnsvgicon icon-add"
        id={undefined}
        onClick={[Function]}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "height": "24px",
            "userSelect": "none",
            "width": "24px",
          }
        }
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
